_source/_posts/2022-05-04-angular-microfrontend-deploy.md

```diff
@@ -0,0 +1,286 @@
+```ts
+import { environment } from '../environments/environment';
+
+const routes: Routes = [
+  { path: '', component: ProductsComponent },
+  {
+    path: 'basket',
+    loadChildren: () => loadRemoteModule({
+      type: 'module',
+      remoteEntry: `${environment.mfe.mfeBasket}/remoteEntry.js`,
+      exposedModule: './Module'
+    }).then(m => m.BasketModule)
+  },
+  {
+    path: 'profile',
+    loadChildren: () => loadRemoteModule({
+      type: 'module',
+      remoteEntry: `${environment.mfe.mfeProfile}/remoteEntry.js`,
+      exposedModule: './Module'
+    }).then(m => m.ProfileModule)
+  },
+  { path: 'login/callback', component: OktaCallbackComponent }
+];
+```
+
+Feel free to serve the project using `npm run run:all` to double-check everything still works.
+
+Now we'll add the configuration for the deployed applications. Open `projects/shell/src/environments/environment.prod.ts` and add the same property for the micro-frontend remotes. Replace the URI with the deployed location.
+
+```ts
+export const environment = {
+  production: true,
+  mfe: {
+    "mfeBasket": "https://{yourVercelDeployPath}.vercel.app",
+    "mfeProfile": "https://{yourVercelDeployPath}.vercel.app"
+  }
+};
+```
+
+Angular automatically replaces the `environment.ts` file to match when you build the project. So if you're serving locally, you'll use the contents from `environment.ts`, and if you run `ng build` to create the release, you'll use the contents from `environment.prod.ts`. 
+
+**Note**: _Vercel has a release promotion mechanism where you first release to a staging environment to conduct release verification, then promote the verified application to prod. With this flow, we'd need to add more complexity to the way we access the URIs. This tutorial will simplify things to only use one environment, but if you need to handle a multi-step release environment configuration, check out the links to blog posts at the end for handling per-environment configuration._
+
+### Add your deployment URL to Okta
+
+You'll need to update the Okta application with the new deploy location so that you can sign in. Open the Okta dashboard. Navigate to **Applications** > **Applications** and select the Okta application you created for this project. On the **General** tab, press the **Edit** button on the **General Settings** section and add the new deploy URI for the `shell` application.
+
+{% img blog/angular-microfrontend-deploy/okta-configure-redirects.jpg alt:"Okta configuration for adding the Vercel deploy URI of shell application as 'Sign-in redirect URIs' and 'Sign-out redirect URIs'. The 'Sign-in redirect URI' should include the '/login/callback' route, while the 'Sign-out redirect URIs' is the deploy URI as is." width:"800" %}{: .center-image }
+
+Next, add the deploy URI as a Trusted Origin in Okta. Navigate to **Security** > **API** and then navigate to the **Trusted Origins** tab. Press **+ Add Origin**. Add the deploy location and select **CORS** for the type.
```

Comment:
is this as intended: "Press *** Add Origin** ? Or is that an extra asterisk?

Reply:
It's a plus sign `Press **+ Add Origin**` to match the text on the button in Vercel